homeassistant/components/mqtt/client.py

```diff
@@ -573,6 +582,9 @@ def _async_queue_subscriptions(
             max_qos = max(qos, self._max_qos.setdefault(topic, qos))
             self._max_qos[topic] = max_qos
             self._pending_subscriptions[topic] = max_qos
+            # Cancel any pending unsubscribe since we are subscribing now
+            if topic in set(self._pending_unsubscribes):
```

Comment:
Do we need to copy the set here?

Reply:
Do we need to cancel any pending subscribes on unsubscribe as well?

Reply:
> Do we need to cancel any pending subscribes on unsubscribe as well?

We already canceling pending subscribes on unsubscribe callback